src/link/MachO/Dylib.zig

```diff
@@ -258,7 +225,7 @@ pub fn parse(self: *Dylib) !void {
         return error.NotDylib;
     }
 
-    const this_arch: Arch = decodeArch(self.header.?.cputype) catch |err| switch (err) {
+    const this_arch: Arch = fat.decodeArch(self.header.?.cputype, true) catch |err| switch (err) {
         error.UnsupportedCpuArchitecture => |e| {
             log.err("unsupported cpu architecture 0x{x}", .{self.header.?.cputype});
```

Comment:
Isn't it that we will duplicate the error log here given that you want `decodeArch` to print the error message in case an error?

Reply:
Yeah - that is just a plain old mistake!